### ISSUE TITLE
Missing download signature public key 

### ISSUE THREAD

**Issue by shishini** [rakudo/rakudo#4642]
## The Problem
The downloads page on rakudo.org https://rakudo.org/downloads/rakudo , have an .asc file for each downloadable file that have a PGP signature 

The public key for those signature is missing, which as I understand makes it not possible to verify the signatures 

## Expected Behavior
Please make the public key available 
Also please if possible add instructions on how to verify the downloads using those signatures

**Comment by shishini** [comment]
This command kind of work, but still not very trustworthy

```
$ gpg --keyserver keyserver.ubuntu.com --auto-key-retrieve --verify rakudo-moar-2021.10-01-win-x86_64-msvc.msi.asc
gpg: WARNING: "--auto-key-retrieve" is a deprecated option
gpg: please use "--keyserver-options auto-key-retrieve" instead
gpg: assuming signed data in `rakudo-moar-2021.10-01-win-x86_64-msvc.msi'
gpg: Signature made Mon, Oct 25, 2021  3:18:23 AM EDT using RSA key ID 5BCDBB01
gpg: requesting key 5BCDBB01 from hkp server keyserver.ubuntu.com
gpg: key 5BCDBB01: public key "Patrick Böker <patrickb@cpan.org>" imported
gpg: Total number processed: 1
gpg:               imported: 1  (RSA: 1)
gpg: Good signature from "Patrick Böker <patrickb@cpan.org>"
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: EF97 6EC6 9E7B B463 4E99  A562 45BE AFAD 5BCD BB01 
```

**Comment by patrickbkr** [comment]
Thanks for the report. I will look into this.

**Comment by jonathanstowe** [comment]
I think the signing key needs to be signed itself by another _trusted_ key, the way to go may be to create e.g. a rakudo.org key and for that to sign the keys (of the releasers,)  FWIW I just signed Patrick's key and sent it back to the key servers.

**Comment by jonathanstowe** [comment]
Assuming that you do indeed _trust_ Patrick you can sign the key yourself to shut that warning up.  e.g.

```
gpg --keyserver keyserver.ubuntu.com --recv-key 5BCDBB01
gpg --sign-key 5BCDBB01
gpg --keyserver keyserver.ubuntu.com  --send-keys 5BCDBB01
```
Maybe we ought to have a "Raku key signing party" :rofl:

**Comment by patrickbkr** [comment]
@jonathanstowe I've been pondering this for a while. I think the solution people expect is to just put the keys on the website. But that won't add much security wrt. someone tampering with the files. Building up a useful WOT around the signing keys does add security. But how do we do that? And how do we communicate about it? Because a WOT only helps if people can and do verify it.

I also wonder if a special Rakudo signing key is a good or bad thing. Having a single key that is just handed around among the releasers indefinitely is obviously a bad idea.

**Comment by patrickbkr** [comment]
A "Raku key signing party" might actually be a good idea. I think most people have seen each other at a conference and could verify the identity in a video call. Then users interested in key integrity could check for those signings.

**Comment by jonathanstowe** [comment]
Yeah having a single signing key for all releasers would be bad as you say, but having a `rakudo.org` key that can be signed by everyone who cares and then signing the key of the releasers would go some way to establishing WOT.

**Comment by vladsf** [comment]
:+1:

**Comment by patrickbkr** [comment]
I have recently set up a more conservative key setup for myself. Thus my keys changed. The new keys are uploaded to https://keys.openpgp.org/ and are signed by my older keys.
Previously I used the identity: `Patrick Böker <patrickb@cpan.org>` the new key has the identity `Patrick Böker (Main key) <patrick.boeker@posteo.de>`. I plan to add the cpan email address to the new key as well, but can't do so at the moment.

As an intermediate step it might make sense to add some text about the cert and checksum files on rakudo.org. I'm a bit unsure what to write though. Just stating that the compiled release files are signed by be and the source release files by jdv is a bit pointless.

**Comment by shishini** [comment]
> As an intermediate step it might make sense to add some text about the cert and checksum files on rakudo.org. I'm a bit unsure what to write though. Just stating that the compiled release files are signed by be and the source release files by jdv is a bit pointless.

The powershell releases page can be used as an example, on how to share checksums
I think section on how to check the signature can also be added 

Just say to check the signature, do steps 1) .. 2).. 3)..

1password, is a commercial product, but they have a nice page on how to verify the download, this also can be an example
https://support.1password.com/verify-download/
